### PR TITLE
DOC-1055: RedisTimeSeries Quick Start

### PR DESCRIPTION
**Pull request by cmilesb**
Staging Link: https://docs.redis.com/staging/jira-doc-1055/modules/redistimeseries/redistimeseries-quickstart/

### CODE REVIEW

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 1]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+### Connect to a database
+
+```sh
+$ redis-cli -h <endpoint> -p <port> -a <password>
+127.0.0.1:12543>
+```
+
+### Create a time series
+
+Use `TS.CREATE` to create a Redis key with a time series value.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:36]: Maybe add a link to the command reference here?
```suggestion
Use [`TS.CREATE`](https://redis.io/commands/ts.create/) to create a Redis key with a time series value.
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 2]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+```sh
+127.0.0.1:12543> TS.CREATE humidity:2:24 LABELS sensor_id 2 area_id 24 data_type "humidity"
+"OK"
+127.0.0.1:12543> TS.CREATE temperature:2:24 LABELS sensor_id 2 area_id 24 data_type "temperature"
+"OK"
+```
+
+### Add new data to a time series
+
+Use `TS.ADD` to add new data to a time series.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:65]:
```suggestion
Use [`TS.ADD`](https://redis.io/commands/ts.add/) to add new data to a time series.
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 3]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+```sh
+127.0.0.1:12543> TS.ADD temperature:1:20 1652824475581 23.9
+1652824475581
+127.0.0.1:12543> TS.ADD temperature:1:20 1652828015127 22.8
+1652828015127
+127.0.0.1:12543> TS.ADD temperature:1:20 1652831657886 21.7
+1652831657886
+```
+
+The second argument in `TS.ADD` is the UNIX timestamp for the sample in milliseconds. You can also specify `*` for the current time on the Redis Server.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:78]:
```suggestion
The second argument in `TS.ADD` is the UNIX timestamp for the sample in milliseconds. You can also specify `*` for the current time on the Redis server.
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 4]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+```sh
+127.0.0.1:12543> TS.ADD humidity:1:20 1652824475581 29
+1652824475581
+127.0.0.1:12543> TS.ADD humidity:1:20 1652828015127 27
+1652828015127
+127.0.0.1:12543> TS.ADD humidity:1:20 1652831657886 28
+1652831657886
+```
+
+You can also update multiple time series at once with `TS.MADD`.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:91]:
```suggestion
You can also update multiple time series at once with [`TS.MADD`](https://redis.io/commands/ts.madd/).
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 5]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+2) "1652824475581"
+3) "1652828015127"
+4) "1652828015127"
+5) "1652831657886"
+6) "1652831657886"
+```
+
+### Get data from a time series
+
+Use `TS.GET` to get the latest entry in the time series.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:109]:
```suggestion
Use [`TS.GET`](https://redis.io/commands/ts.get/) to get the latest entry in the time series.
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 6]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+
+```sh
+127.0.0.1:12543> TS.GET temperature:1:20
+1) "1652831657886"
+2) "21.7"
+```
+
+Use `TS.RANGE` to get a range of timestamps and values in the time series, ordered from earliest to latest. `TS.REVRANGE` orders the values from latest to earliest.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:117]:
```suggestion
Use [`TS.RANGE`](https://redis.io/commands/ts.range/) to get a range of timestamps and values in the time series, ordered from earliest to latest. [`TS.REVRANGE`](https://redis.io/commands/ts.revrange/) orders the values from latest to earliest.
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 7]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+127.0.0.1:12543> TS.REVRANGE temperature:2:24 0 +
+1) 1) "1652831657886"
+   2) "16.7"
+2) 1) "1652828015127"
+   2) "17.2"
+3) 1) "1652824475581"
+   2) "18.3"
+```
+
+Use `TS.MGET` to get the last entry in all time series keys with an area_id of 20.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:136]:
```suggestion
Use [`TS.MGET`](https://redis.io/commands/ts.mget/) to get the last entry in all time series keys with an `area_id` of 20.
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 8]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+         2) "1"
+      2) 1) "area_id"
+         2) "20"
+      3) 1) "data_type"
+         2) "temperature"
+   3) 1) "1652831657886"
+      2) "21.7"
+```
+
+Use `TS.MRANGE` to get a range of entries in all time series keys with a sensor_id of 2 from earliest to latest. Then, use `TS.MREVRANGE` to get a range of entries in all time series keys that are humidity keys from latest to earliest.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:160]: You might also consider breaking these two examples up for clarity.
```suggestion
Use [`TS.MRANGE`](https://redis.io/commands/ts.mrange/) to get a range of entries in all time series keys with a `sensor_id` of 2 from earliest to latest. Then, use [`TS.MREVRANGE`](https://redis.io/commands/ts.mrevrange/) to get a range of entries in all time series keys that are humidity keys from latest to earliest.
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 9]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+2) 1) "1652828015127"
+   2) "17.2"
+127.0.0.1:12543> TS.REVRANGE temperature:2:24 0 +
+1) 1) "1652831657886"
+   2) "16.7"
+2) 1) "1652828015127"
+   2) "17.2"
+3) 1) "1652824475581"
+   2) "18.3"
+```
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:134]: Maybe break up these examples and their descriptions for clarity.

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 10]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+         2) "56"
+      2) 1) "1652828015127"
+         2) "54"
+      3) 1) "1652824475581"
+         2) "51"
+```
+
+### Delete data from a time series
+
+Use TS.DEL to delete all timestamps in a specified range from a time series.
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:201]:
```suggestion
Use [`TS.DEL`](https://redis.io/commands/ts.del/) to delete all timestamps in a specified range from a time series.
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 11]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+127.0.0.1:12543> TS.DEL temperature:1:20 1652824475581 1652828015127
+(integer) 2
+127.0.0.1:12543> TS.RANGE temperature:1:20 0 +
+1) 1) "1652831657886"
+   2) "21.7"
+```
+
+## RedisTimeSeries with Python
+
+If you want to use RedisTimeSeries within an application, you can use one of the [client libraries](https://redis.io/docs/stack/json/clients/).
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:213]:
```suggestion
If you want to use RedisTimeSeries within an application, you can use one of the [client libraries](https://redis.io/docs/stack/timeseries/clients/).
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 12]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+
+The following example uses the Redis Python client library [redis-py](https://github.com/redis/redis-py), which supports RedisTimeSeries commands as of v4.0.0.
+
+This Python code adds multiple time series keys to Redis, adds data to the time series, retrieves data from the time series, and then deletes the time series keys.
+
+```python
+import redis
+import json #For making the responses easier to read
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:221]:
```suggestion
import json # To make the responses easier to read
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 13]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+
+# JSON-like data from two sensors
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:223]:
```suggestion
# Data from two sensors
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 14]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+}
+
+sensors = [ sensor1, sensor2 ]
+
+# Connect to a Redis database
+r = redis.Redis(host="<endpoint>", port="<port>",
+    password="<password>")
+
+key_list = [] #Creating a list of keys for later
+humidity_madd_list = [] #Creating a list of keys & values for madd
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:255]:
```suggestion
# Create a list to store keys
key_list = []
# Create a list of keys and values for ts.madd
humidity_madd_list = []
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 15]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+    # Add temperature data to time series keys with ts.add
+    print("Add data to {}".format(temperature_key))
+    # For each data entry:
+    for entry in sensor["data"]:
+        r.ts().add(temperature_key, entry["timestamp"], entry["temperature"])
+        #Create a list of humidity data for ts.madd
+        humidity_madd_list.append( (humidity_key, entry["timestamp"], entry["humidity"]))
+
+#Add humidity data to time series keys with ts.madd
+print("Add data with madd:\n{}".format(humidity_madd_list))
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:283]:
```suggestion
print("Add data with ts.madd:\n{}".format(humidity_madd_list))
```

File: content/modules/redistimeseries/redistimeseries-quickstart.md [hunk 16]
```diff
@@ -9,11 +9,533 @@ aliases: /rs/getting-started/creating-database/redistimeseries/
...
+
+### Example output
+
+```sh
+$ python3 quick_start.py
+Create temperature-py:1:20 and humidity-py:1:20
+Add data to temperature-py:1:20
+Create temperature-py:2:24 and humidity-py:2:24
+Add data to temperature-py:2:24
+Add data with madd:
```
Comment (rrelledge) [inline content/modules/redistimeseries/redistimeseries-quickstart.md:330]:
```suggestion
Add data with ts.madd:
```

**Review by nermiller — APPROVED** [review]
LGTM

**Review by rrelledge — CHANGES_REQUESTED** [review]
Looks good so far. I just left some suggestions.

**Pull request merged**